### PR TITLE
chore: cleanup content types

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Clean up content types as the experiment won.
- Introduce a new component called `CardCheckbox` inspired by the CustomCheckbox from this initiative.

Preview:

https://github.com/user-attachments/assets/1745573a-4e11-48eb-8ab9-fb27c238f84e

The borders look a little bit weird from the recording above, but doing a screenshot reflects the actual colors like below:

Screenshot 2024-08-04 at 5 55 07 AM

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

MI-451

### CODE REVIEW

File: packages/shared/src/components/typography/Typography.tsx [hunk 1]
```diff
@@ -62,6 +63,7 @@ export function Typography<TagName extends AllowedTags>({
   className,
   ...props
 }: TypographyProps<TagName>): ReactElement {
-  const Tag = classed(tag, className, type, bold && 'font-bold', color);
+  const classes = classNames(className, type, bold && 'font-bold', color);
+  const Tag = classed(tag, classes);
```
Comment (sshanzel) [inline packages/shared/src/components/typography/Typography.tsx:67]: There's an issue with the `classed` component when the first parameter `className` is null or undefined. If that is the case, the following classes are not being applied.

File: packages/shared/src/components/fields/CardCheckbox.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import React, { InputHTMLAttributes, ReactElement } from 'react';
```
Comment (sshanzel) [inline packages/shared/src/components/fields/CardCheckbox.tsx:1]: Things would have been simpler without the gradient borders 😒

File: packages/shared/src/components/typography/Typography.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/typography/Typography.tsx:68]: wow good find!
CC: @capJavert just so you know, maybe this was the case 🤯

**Review by rebelchris — APPROVED** [review]
Looks good to me!

**Review by omBratteng — APPROVED** [review]

File: packages/shared/src/components/typography/Typography.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/typography/Typography.tsx:68]: Didn't want to touch the `classed` file yet, but we can maybe write a ticket for it if anyone wants 😅

**Pull request merged**